src/activity_handlers.cpp

```diff
@@ -1632,7 +1632,7 @@ void activity_handlers::repair_item_finish( player_activity *act, player *p )
         return;
     }
 
-    item &fix = p->i_at( act->position );
+    item &fix = *act->targets[0];
```

Comment:
`act->targets.front()` would be cleaner. Also, it wouldn't hurt to have an `assert` here since `targets` may be empty.